### ISSUE TITLE
Docker: Remove incorrect permission check for storage folder on startup

### ISSUE THREAD

**Issue by lastzero** [photoprism/photoprism#2334]
See https://github.com/photoprism/photoprism/discussions/2328#discussioncomment-2771617:

I have been made some test and is a problem with the debian bookworm version.
My test:
I pulled and run an interative sesion to a bare debian:bookworm:
```
sudo docker run -i -t debian:bookworm bash
```
Then just move to the home and create the audit.sh file with the following data:
```
#!/bin/bash

STORAGE_PATH=${PHOTOPRISM_STORAGE_PATH:-storage}

set -e

# create directory if not exists
mkdir -p "${STORAGE_PATH}" || (echo "Failed creating storage folder \"$STORAGE_PATH\", see $DOC_URL" 1>&2; exit 1)

# check directory permissions
[[ -w "${STORAGE_PATH}" ]] || \
  (echo "Storage folder \"$STORAGE_PATH\" is not writable, see $DOC_URL" 1>&2; exit 1)

# create and delete test file
(touch "${STORAGE_PATH}/is-writable" 2>/dev/null && rm "${STORAGE_PATH}/is-writable") || \
  (echo "Failed creating test file in storage folder, see $DOC_URL" 1>&2; exit 1)
```
Then chmod a+x audit.sh and run it

It fails with the permission error.

However if I made the exact same test inside a bullseye that I also fetch with:

```
sudo docker run -i -t debian:bullseye bash
```

It works without problem:
```
root@37d540302cd8:/home#chmod a+x audit.sh
root@37d540302cd8:/home# ./audit.sh
root@37d540302cd8:/home# ls
audit.sh  storage
root@37d540302cd8:/home#

```

**Comment by timespacedecay** [comment]
The recent update a few days ago causes Photoprism to not start on Unraid, and it looks like its related to this storage permission check for bookworm:

```
started 220517-bookworm as root (amd64-prod)
init: updating filesystem permissions
PHOTOPRISM_DISABLE_CHOWN="true" disables permission updates
Problems? Our Troubleshooting Checklists help you quickly diagnose and solve them:
https://docs.photoprism.app/getting-started/troubleshooting/
file umask....: "0002" (u=rwx,g=rwx,o=rx)
home directory: /photoprism
assets path...: /opt/photoprism/assets
storage path..: /photoprism/storage
config path...: default
cache path....: default
backup path...: /photoprism/storage/backups
import path...: /photoprism/import
originals path: /photoprism/originals
switching to uid 99:100
/opt/photoprism/bin/photoprism start
Storage folder "/photoprism/storage" is not writable, see https://docs.photoprism.app/getting-started/troubleshooting/docker/#file-permissions
```

**Comment by gpulido** [comment]
Hello,
The last patch solve the problem. I can now use the preview tag

**Comment by fabianhick** [comment]
I unfortunately still experience this error both on the `latest` and the `preview` tag. 
I suppose it is related to this check, as I provided all permissions to the folders created by docker.
However, my errors might also be related to the following unsolved thread: https://www.reddit.com/r/photoprism/comments/u7ee0e/initial_setup_permission_denied/

**Comment by lastzero** [comment]
Did you check these recommendations? https://docs.photoprism.app/getting-started/troubleshooting/docker/#file-permissions

Does it work with the Ubuntu jammy base image? See Docker Hub.

**Comment by fabianhick** [comment]
Yes, I did check those and also generously allowed permissions, even though the folders have already been created by docker, as they didn't exist before.
Sadly, neither the jammy=based images worked.

```
fabian@fedora ~/d/photoprism> sudo docker-compose up
Starting pprism_photoprism_1 ... done
Attaching to pprism_photoprism_1
photoprism_1  | started 220730 as root (amd64-prod)
photoprism_1  | Problems? Our Troubleshooting Checklists help you quickly diagnose and solve them:
photoprism_1  | https://docs.photoprism.app/getting-started/troubleshooting/
photoprism_1  | file umask....: "0002" (u=rwx,g=rwx,o=rx)
photoprism_1  | home directory: /photoprism
photoprism_1  | assets path...: /opt/photoprism/assets
photoprism_1  | storage path..: /photoprism/storage
photoprism_1  | config path...: default
photoprism_1  | cache path....: default
photoprism_1  | backup path...: /photoprism/storage/backups
photoprism_1  | import path...: /photoprism/import
photoprism_1  | originals path: /photoprism/originals
photoprism_1  | running as uid 0
photoprism_1  | /opt/photoprism/bin/photoprism start
photoprism_1  | Failed creating test file in storage folder, see https://docs.photoprism.app/getting-started/troubleshooting/docker/#file-permissions
pprism_photoprism_1 exited with code 0
```

```
fabian@fedora ~/d/photoprism> ls -lah
total 12K
drwxrwxrwx. 1 fabian fabian   66 Jul 31 02:02 ./
drwxrwxr-x. 1 fabian fabian   32 Jul 30 19:33 ../
-rwxrwxrwx. 1 fabian fabian 9.3K Jul 31 02:02 docker-compose.yml*
drwxrwxrwx. 1 root   root      0 Jul 31 02:02 Pictures/
drwxrwxrwx. 1 root   root      0 Jul 31 02:02 storage/
```
excerpt from docker inspect:
```
        "Mounts": [
            {
                "Type": "bind",
                "Source": "/home/fabian/docker/photoprism/Pictures",
                "Destination": "/photoprism/originals",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/home/fabian/docker/photoprism/storage",
                "Destination": "/photoprism/storage",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            }
        ],
```

**Comment by lastzero** [comment]
These folders belong root but you start Docker as a non root User? Maybe that's the problem. The permission check is made in the binary using a system call. I would have to debug deeper into it, but we are on vacation now.

**Comment by lastzero** [comment]
Try running PhotoPrism as a non root users (with your UID) and see if that works. Are there any other layers of virtualization, like Windows or a VM?

**Comment by lastzero** [comment]
@fabianhick I recommend you open a [GitHub Discussion](https://github.com/photoprism/photoprism/discussions) for this and provide complete information about your environment, so it's possible to provide you with helpful advice.

This Google search finds about 88,300 results, which may give you a clue as to why the syscall is failing:

- https://www.google.com/search?q=fedora+syscalls+docker&oq=fedora+syscalls+docker

In particular, check the Docker version you are using and make sure Linux kernel security is either disabled or properly configured.

**Comment by fabianhick** [comment]
Thanks, got it resolved by further research: Even though I thought all kernel restrictions would be suspended by the `security_opt` settings, SELinux still restricted the usage of the mounted volumes.
TLDR: The solution was to append `:z` to the volume mounts as mentioned [here](https://docs.docker.com/storage/bind-mounts/#configure-the-selinux-label). 
Might be worth to add to the checklist.

**Comment by lastzero** [comment]
Feel free to send a PR for our Troubleshooting Guide and/or Getting Started FAQ.
